### PR TITLE
feat: TE7 - T1040: IOS-KT(Components)

### PR DESCRIPTION
**Pull request by ShubhamWorks02**
Created Demo which shows User Document and a form to fill for information.

### Key Features

- Implemented singleton pattern for dismissing keyboard
- Model which uses sectionData according to which is provided and shows document which adapts to dynamic data(scalable).
- Created custom Designable classes for textfields
- Form Details validation
- can get Document approval status which changes Document Style according to status

### CODE REVIEW

File: UiBasic/ComponentsKT/ComponentsKT/ViewControllers/FormViewController.swift [hunk 1]
```diff
@@ -39,105 +38,105 @@ class FormViewController: UIViewController {
...
+        cameraImg.image = UIImage(named: Constants.Strings.imgCameraStr)
         cameraImg.frame = CGRect(x: childView.bounds.midX - 6.5, y: childView.bounds.midY - 6.5, width: 13, height: 13)
         childView.layer.borderWidth = 2
         childView.layer.borderColor = UIColor.white.cgColor
         cameraImg.layer.cornerRadius = cameraImg.bounds.width / 2
-
         childView.clipsToBounds = true
         cameraImg.isUserInteractionEnabled = true
         childView.isUserInteractionEnabled = true
         imgProfileView.isUserInteractionEnabled = true
```
Comment (iOSDev-Auction) [inline UiBasic/ComponentsKT/ComponentsKT/ViewControllers/FormViewController.swift:88]: why isUserInteractionEnabled for all? is there any specific requirement

File: UiBasic/ComponentsKT/ComponentsKT/ViewControllers/FormViewController.swift [hunk 2]
```diff
@@ -39,105 +38,105 @@ class FormViewController: UIViewController {
...
+        let cameraImg = UIImageView()
+        cameraImg.image = UIImage(named: Constants.Strings.imgCameraStr)
         cameraImg.frame = CGRect(x: childView.bounds.midX - 6.5, y: childView.bounds.midY - 6.5, width: 13, height: 13)
         childView.layer.borderWidth = 2
         childView.layer.borderColor = UIColor.white.cgColor
         cameraImg.layer.cornerRadius = cameraImg.bounds.width / 2
-
         childView.clipsToBounds = true
         cameraImg.isUserInteractionEnabled = true
         childView.isUserInteractionEnabled = true
```
Comment (shilpeshsimform) [inline UiBasic/ComponentsKT/ComponentsKT/ViewControllers/FormViewController.swift]: Why isUserInteractionEnabled allowed for multiple components?,is there any specific requirement?

File: UiBasic/ComponentsKT/ComponentsKT/ViewControllers/ViewController.swift [hunk 3]
```diff
@@ -49,26 +49,29 @@ class ViewController: UIViewController {
...
+
+// MARK: ACTION METHODS
+extension ViewController {
+    
+    
     @IBAction func btnBellPressed(_ sender: UIButton) {
-        let storyboard = UIStoryboard(name: "Main", bundle: nil)
-        let destinationVC = storyboard.instantiateViewController(withIdentifier: "FormViewController") as! FormViewController
+        let storyboard = UIStoryboard(name: Constants.StoryBoards.main, bundle: nil)
+        let destinationVC = storyboard.instantiateViewController(withIdentifier: Constants.VCs.formViewController) as! FormViewController
```
Comment (shilpeshsimform) [inline UiBasic/ComponentsKT/ComponentsKT/ViewControllers/ViewController.swift]: Why VC is forcefully wrapped?

File: UiBasic/ComponentsKT/ComponentsKT/ViewControllers/FormViewController.swift (hunk 2 above)
Comment (ShubhamWorks02) [reply UiBasic/ComponentsKT/ComponentsKT/ViewControllers/FormViewController.swift]: view heirarchy is situated on top of another so it is needed but childview's user interaction is not needed, will change it

File: UiBasic/ComponentsKT/ComponentsKT/ViewControllers/ViewController.swift (hunk 3 above)
Comment (ShubhamWorks02) [reply UiBasic/ComponentsKT/ComponentsKT/ViewControllers/ViewController.swift]: it was done by mistake, will change it

**Comment by ShubhamWorks02** [comment]
changes done @shilpeshsimform

**Pull request merged**